contrib/platform/test/com/sun/jna/platform/win32/Crypt32Test.java

```diff
@@ -61,5 +71,12 @@ public void testCryptProtectUnprotectDataWithEntropy() {
     	Kernel32.INSTANCE.LocalFree(pDataEncrypted.pbData);
     	Kernel32.INSTANCE.LocalFree(pDataDecrypted.pbData);
     	Kernel32.INSTANCE.LocalFree(pDescription.getValue());
-    }    
+    }  
+
+	public void testCertAddEncodedCertificateToSystemStore() {
+		// try to install a non-existent certificate
+		assertFalse(Crypt32.INSTANCE.CertAddEncodedCertificateToSystemStore("ROOT", null, new DWORD(0)));
+		// should fail with "unexpected end of data"
+		assertEquals(WinCrypt.CRYPT_E_ASN1_EOD, Native.getLastError());
+	}
```

Comment:
The comments should be the assertion messages. While it is true that if the assertion fails one can navigate to the code and see the comment, but why not make this plain from the failure message:
```java
assertFalse("Unexpected success to install non-existent certificate", ...);
assertEquals("Unexpected failure code", ..., ...);
```